src/components/buttons/ScrollableButton/index.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  Animated,
+  LayoutChangeEvent,
+  StyleProp,
+  TextStyle,
+  View,
+  ViewStyle,
+} from 'react-native';
+
+import { Touchable } from '@/components/common';
+import AddGradient from '@/icons/svg/AddGradient.svg';
+import { Nullable } from '@/interfaces/general';
+
+import styles from './styles';
+
+interface Props {
+  onPress: () => void;
+  text: string;
+  buttonStyle?: StyleProp<ViewStyle>;
+  imageStyle?: StyleProp<ViewStyle>;
+  textStyle?: StyleProp<TextStyle>;
+  scrollPosition: number;
+}
+
+function ScrollableButton({
+  onPress,
+  text,
+  textStyle,
+  buttonStyle,
+  imageStyle,
+  scrollPosition,
+  ...props
+}: Props) {
+  const [showFullButton, setShowFullButton] = useState(true);
+  const [animationFinished, setAnimationFinished] = useState(true);
+  const [currentScrollPosition, setCurrentScrollPosition] = useState(0);
+  const [textWidth, setTextWidth] = useState<Nullable<number>>(null);
```

Comment:
We can avoid nullable by doing this:
```suggestion
  const [textWidth, setTextWidth] = useState<number>();
```
and the state accepts `number | undefined`

Reply:
great!